waddrmgr/db.go

```diff
@@ -695,12 +695,14 @@ func deleteAccountIDIndex(tx walletdb.Tx, account uint32) error {
 // putAccountNameIndex stores the given key to the account name index of the database.
 func putAccountNameIndex(tx walletdb.Tx, account uint32, name string) error {
 	bucket := tx.RootBucket().Bucket(acctNameIdxBucketName)
-
-	// Write the account number keyed by the account name.
-	err := bucket.Put(stringToBytes(name), uint32ToBytes(account))
-	if err != nil {
-		str := fmt.Sprintf("failed to store account name index key %s", name)
-		return managerError(ErrDatabase, str, err)
+	// Skip if we're working with the initial db version
```

Comment:
Why would this function ever be run if the DB was not upgraded to the latest version?